### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by lucas-crodrigues**
This is the first part of the Blog app initiated in the first week of the Rails module in the Microverse program.

The requirements are: 

> ### Project requirements
> Be sure to reference the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) from the sneak peek if you have any questions about table or column names or data relationships.
> 
> - Create a new app. Give it a meaningful name.
> - Initialize your project with Git.
> - Make sure that your project has Postgres database set up.
> - Build your project schema.
>     - Create and run the necessary migration files.
>     - Table and column names should match the ERD diagram.
>         - note: photo for users table should be a link to an image
>     - Foreign keys should be included.
>     - All columns that are foreign keys should have a corresponding index.

- [x] Create a new app.
- [x] Initialize the project with Git.
- [x] Postgres database set up.
- [x] Build project schema.

Thank you.

### CODE REVIEW

File: db/migrate/20221019161546_add_user_to_posts.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
+  end
+end
```
Comment (medaminedev66) [inline db/migrate/20221019161546_add_user_to_posts.rb]: - [ ] The reference is not directly `user`, it is instead `author` to table `user`. Kindly see the below example on how to make such a migration for adding a reference with a different name than the reference model class name, and make changes in your database. :pray: 

image

File: db/migrate/20221019161810_add_user_to_comments.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
+  end
+end
```
Comment (medaminedev66) [inline db/migrate/20221019161810_add_user_to_comments.rb]: - [ ] Again, the reference name should be `author`. Please refer to the above similar comment for the other migration. :pray:

File: db/migrate/20221019161830_add_user_to_likes.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
+  end
+end
```
Comment (medaminedev66) [inline db/migrate/20221019161830_add_user_to_likes.rb]: - [ ] Again, the reference name should be `author`. Please refer to the above similar comment for the other migration. :pray:

File: db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
+  add_foreign_key "comments", "posts"
+  add_foreign_key "comments", "users"
+  add_foreign_key "likes", "posts"
+  add_foreign_key "likes", "users"
+  add_foreign_key "posts", "users"
+end
```
Comment (medaminedev66) [inline db/schema.rb]: After applying the requested changes in the migrations, this should look like: 

image

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @lucas-crodrigues  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- GitFlow best practices have been followed.
- There are no linter errors.

## Required Changes ♻️
- [ ] **Your readme isn't descriptive, and it doesn't provide enough details on the project. Please update your Readme with the proper one. . You can use the [Microver template](https://github.com/microverseinc/readme-template) as an example.**

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20221019161546_add_user_to_posts.rb (hunk 1 above)
Comment (rloterh) [reply db/migrate/20221019161546_add_user_to_posts.rb]: well resolved! ✔️

File: db/migrate/20221019161810_add_user_to_comments.rb (hunk 2 above)
Comment (rloterh) [reply db/migrate/20221019161810_add_user_to_comments.rb]: well resolved! ✔️

File: db/migrate/20221019161830_add_user_to_likes.rb (hunk 3 above)
Comment (rloterh) [reply db/migrate/20221019161830_add_user_to_likes.rb]: well resolved! ✔️

File: db/schema.rb (hunk 4 above)
Comment (rloterh) [reply db/schema.rb]: well resolved! ✔️

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Lucas, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Data model properly implemented

## Required Changes ♻️
- [ ] This repository does not have an appropriate ReadMe. Kindly ensure that your ReadMe to this project is detailed and reflects the project. Kindly use this [recommended template](https://github.com/microverseinc/readme-template/blob/master/README.md), just as mentioned in the previous review.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lucas-crodrigues** [comment]
What a blunder! Forgetting the readme altogether, oh well, thank you for the feedback @rloterh and @medaminedev66, cheers 🤙

**Review by DammyShittu — APPROVED** [review]
Hi @lucas-crodrigues,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**